### ISSUE TITLE
Better amplitude results for single frequency

### ISSUE THREAD

**Issue by matthewrankin** [questrail/siganalysis#14]
Instead of simply returning the values of a single frequency bin, for frequencies that are off center a frequency bin, should I return a weighted-average of the adjacent bins?

For instance, if the FFT results have a frequency step size of 3 Hz starting at 0 Hz, and I'm wanting to know the amplitude of a signal at 1 Hz, should I return the weighted average of the amplitude found in the 0 Hz bin and the 3 Hz bin?

By weighted-average, I'm thinking in this case to return two-thirds amplitude from the 0 Hz bin and one-third amplitude from the 3 Hz bin.

**Comment by matthewrankin** [comment]
Tested the proposal. It does not work — the weighted average is **worse** than the single bin it would replace across most of the range, and the window choice added in #6 solves the problem outright.

## Measurements

A 1.0 amplitude tone swept across one 10 Hz bin (1 kHz sampling, 0.1 sec frames), `offset` being how far the tone sits from the center of a bin in bin widths, so 0.5 is exactly between two bins:

| offset | nearest bin (current) | weighted average (proposed) | flattop |
|---|---|---|---|
| 0.00 | 1.0000 | 1.0000 | 1.0000 |
| 0.10 | 0.9914 | 0.9435 | 1.0001 |
| 0.20 | 0.9679 | 0.8930 | 1.0002 |
| 0.30 | 0.9309 | 0.8536 | 1.0003 |
| 0.40 | 0.8817 | 0.8285 | 1.0000 |
| 0.50 | 0.8176 | 0.8196 | 0.9990 |
| 0.60 | 0.8788 | 0.8279 | 1.0000 |
| 0.70 | 0.9300 | 0.8533 | 1.0003 |
| 0.80 | 0.9686 | 0.8935 | 1.0002 |
| 0.90 | 0.9924 | 0.9443 | 1.0001 |

Worst-case error over the sweep:

- nearest bin, hamming — **18.24%**
- weighted average, hamming — **18.04%**
- nearest bin, flattop — **0.10%**

## Why the weighted average fails

The proposal reasons about bins as if each held the energy of a rectangular slice of spectrum, so that a tone between two bins splits its energy proportionally between them. That is not what a bin holds. A bin holds the signal convolved with the window's frequency response, and both neighbors of an off-center tone are already attenuated by that response. Averaging two attenuated readings produces a third, lower reading rather than recovering the original amplitude — which is why the proposed column sits *below* the current one almost everywhere. It only draws level at offset 0.5, where both bins are attenuated equally and the average is the least wrong it can be. It buys 0.2 percentage points at the one offset it was designed for and gives up as much as 7.5 points elsewhere.

## What does work

Scalloping loss is a property of the window, so the fix belongs at the window, not at the readout. `flattop`, now available through `stft(..., window="flattop")` after #6, is the window a spectrum analyzer offers for exactly this measurement: its main lobe is deliberately flat across the top so a tone anywhere in the bin reads its true amplitude. It holds within 0.1% across the whole sweep against 18% for the current default. The cost is frequency resolution — the wide main lobe smears energy across neighboring bins, so it is the wrong choice when resolving two close tones matters and the right one when amplitude accuracy matters.

## Recommendation

Close this in favor of documenting the tradeoff, since the code change it asks for is a regression and the capability it wants already exists. Worth adding to the `single_frequency_over_time()` docstring: the amplitude it reports is only as accurate as the window allows, and `window="flattop"` is the answer for a frequency that does not fall on a bin center.

The one case still not covered is not knowing the frequency in advance and wanting to estimate a peak between bins. That is a different problem with a different solution (Jacobsen or quadratic peak interpolation on the log magnitudes), and worth its own issue if it ever comes up.
